### PR TITLE
Feature/40791

### PR DESCRIPTION
**Pull request by sa11616079**
1. Setup the Material UI theme with font-family as ‘"Comic Sans MS", cursive, sans-serif’, and htmlFontSize as 10 for typography.
Refer to https://material-ui.com/customization/themes/
2. Create a theme.js file in “src” folder.
3. Setup the theme in “App.js” using above created theme.js file.
4. Now create a component named as “Math” that will accept following props:
* first - a number and required
* second - a number and required
* operator - must be any of ‘+’, ‘-’, ‘/’, ‘*’
* children - optional and must contain function that will be used to render the template provided by the user, and will accept the object with keys, first, second, operator, and result. This will used to provide the template to the Math component as children.
5. Please note in case of “/” operator if second value is 0 then result must be “Infinity”
6. If anyhow operator must not be in the provide list then result must be “Invalid Operator”.
7. Default template is:
- 7 + 4 = 11
- 7 - 3 = 4
- 7 / 0 = Infinity
- 7 ^ 8 = Invalid Operation.
8. Using children we can provide various templates to the component and can print the result in any format;
- Sum of 7 and 4 is 11
- When we add 3 with 4 then we will get 7 as result.
- and so on.
9. Show the various usage of Math Component in “ChildrenDemo” component. It will be similar to another Demo component.
10. One of the user-defined templates must use the “Typography” component from “Material UI”.

MUI

### CODE REVIEW

File: training/src/components/Math/Math.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        {first}
+        {' '}
+        {operator}
+        {' '}
+        {second}
+        {' '}
+        =
+        {' '}
+        {result}
+        {' '}
```
Comment (SHUBHAM-JAIN-SS) [inline training/src/components/Math/Math.jsx:35]: use string interpolation

File: training/src/pages/ChildrenDemo/ChildrenDemo.jsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                {first}
+                {' '}
+                and
+                {' '}
+                {second}
+                {' '}
+                is equal to
+                {' '}
+                {result}
+                {' '}
```
Comment (SHUBHAM-JAIN-SS) [inline training/src/pages/ChildrenDemo/ChildrenDemo.jsx:35]: string interpolation

File: training/src/pages/ChildrenDemo/ChildrenDemo.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                  {result}
+                  {' '}
+                </p>
+              )
+            }
+          </Text>
+        </ThemeProvider>
+      </>
+    );
+  }
```
Comment (SHUBHAM-JAIN-SS) [inline training/src/pages/ChildrenDemo/ChildrenDemo.jsx:64]: use typography component from MUI in one example

File: training/src/pages/ChildrenDemo/ChildrenDemo.jsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                {' '}
+                is equal to
+                {' '}
+                {result}
+                {' '}
+              </p>
+            )
+          }
+        </Text>
+        <ThemeProvider theme={Theme}>
```
Comment (SHUBHAM-JAIN-SS) [inline training/src/pages/ChildrenDemo/ChildrenDemo.jsx:40]: Theme Provider should have been in the app.jsx unless told to you otherwise

**Review by SHUBHAM-JAIN-SS — CHANGES_REQUESTED** [review]

File: training/src/pages/ChildrenDemo/ChildrenDemo.jsx (hunk 4 above)
Comment (sa11616079) [reply training/src/pages/ChildrenDemo/ChildrenDemo.jsx:21]: Theme Provider would be in childrenDemo.jsx file as told by trainer

Comment (SHUBHAM-JAIN-SS) [reply training/src/pages/ChildrenDemo/ChildrenDemo.jsx:21]: ok

File: training/src/pages/ChildrenDemo/ChildrenDemo.jsx (hunk 3 above)
Comment (SHUBHAM-JAIN-SS) [reply training/src/pages/ChildrenDemo/ChildrenDemo.jsx:45]: what about this?

File: training/src/pages/ChildrenDemo/ChildrenDemo.jsx (hunk 2 above)
Comment (SHUBHAM-JAIN-SS) [reply training/src/pages/ChildrenDemo/ChildrenDemo.jsx]: like: `Sum of ${first} ....`

**Pull request merged**